pkg/rpc/server/listen.go

```diff
@@ -121,7 +121,11 @@ func newDBWorker(dbClient dbFile.Operation) dbWorker {
 }
 
 func (w dbWorker) update(ctx context.Context, appVersion, cacheDir string,
-	dbUpdateWg, requestWg *sync.WaitGroup) error {
+	skipDbUpdate bool, dbUpdateWg, requestWg *sync.WaitGroup) error {
+	if skipDbUpdate { // don't update db if `skip-db-update` flag is enabled
```

Comment:
What if just passing the value to `NeedsUpdate`?

```
	needsUpdate, err := w.dbClient.NeedsUpdate(appVersion, skipDBUpdate)
```

Reply:
Done